File: Examples/CSharp/AsposePDF/Forms/TextBox.cs
//////////////////////////////////////////////////////////////////////////
// Copyright 2001-2013 Aspose Pty Ltd. All Rights Reserved.
//
// This file is part of Aspose.Pdf. The source code in this file
// is only intended as a supplement to the documentation, and is provided
// "as is", without warranty of any kind, either expressed or implied.
//////////////////////////////////////////////////////////////////////////
using System.IO;

using Aspose.Pdf;
using Aspose.Pdf.InteractiveFeatures.Forms;
using Aspose.Pdf.InteractiveFeatures.Annotations;

namespace CSharp.AsposePdf.Forms
{
    public class TextBox
    {
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_AsposePdf_Forms();

            //Open document
            Document pdfDocument = new Document(dataDir + "TextBox.pdf");

            //Create a field
            TextBoxField textBoxField = new TextBoxField(pdfDocument.Pages[1], new Aspose.Pdf.Rectangle(100, 200, 300, 300));
            textBoxField.PartialName = "textbox1";
            textBoxField.Value = "Text Box";
            
            //textBoxField.Border = new Border(
            Border border = new Border(textBoxField);
            border.Width = 5;
            border.Dash = new Dash(1, 1);
            textBoxField.Border = border;

            textBoxField.Color = Aspose.Pdf.Color.FromRgb(System.Drawing.Color.Green);

            //Add field to the document
            pdfDocument.Form.Add(textBoxField, 1);

            //Save modified PDF
            pdfDocument.Save(dataDir + "TextBox_out.pdf");

        }
    }
}

File: Examples/VisualBasic/AsposePDF/Forms/TextBox.vb
Imports System.IO
Imports System
Imports Microsoft.VisualBasic
Imports Aspose.Pdf.Forms
Imports Aspose.Pdf
Imports Aspose.Pdf.Annotations

Namespace AsposePDF.Forms
    Public Class TextBox
        Public Shared Sub Run()
            ' ExStart:AddTextBoxField
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_AsposePdf_Forms()

            'Open document
            Dim pdfDocument As New Document(dataDir & Convert.ToString("TextField.pdf"))

            ' Create a field
            Dim textBoxField As New TextBoxField(pdfDocument.Pages(1), New Rectangle(100, 200, 300, 300))
            textBoxField.PartialName = "textbox1"
            textBoxField.Value = "Text Box"

            ' TextBoxField.Border = new Border(
            Dim border As New Border(textBoxField)
            border.Width = 5
            border.Dash = New Dash(1, 1)
            textBoxField.Border = border

            textBoxField.Color = Color.FromRgb(System.Drawing.Color.Green)

            ' Add field to the document
            pdfDocument.Form.Add(textBoxField, 1)

            dataDir = dataDir & Convert.ToString("TextBox_out_.pdf")
            ' Save modified PDF
            pdfDocument.Save(dataDir)
            ' ExEnd:AddTextBoxField
            Console.WriteLine(Convert.ToString(vbLf & "Textbox field added successfully." & vbLf & "File saved at ") & dataDir)

        End Sub
    End Class
End Namespace
